getspice.py

```diff
@@ -22,135 +20,222 @@
+            self._download()
 
-def attemptSpiceDownloadXTimes(x, root_url, result, root_dir):
-    attempts = 0
-    while attempts < x:
-        try:
-            print("Attempting to download spice...")
-            download = downloadSpice(root_url + result)
+            # Unpack the file
             print('Unpacking... (this may take some time!)')
-            if result.endswith('zip'):
-                filelike = io.BytesIO(download.read())
-                with zipfile.ZipFile(filelike, 'r') as archive:
-                    archive.extractall(root_dir)
-                filelike.close()
-            else:
-                cmd = 'gunzip | tar xC ' + root_dir
-                proc = subprocess.Popen(cmd, shell=True, stdin=subprocess.PIPE)
-                proc.stdin.write(download.read())
-            download.close()
-            break
-        except urllib.error.URLError as err:
-            print("Download failed with URLError: {}, trying again after 15 seconds!".format(err.reason))
-        attempts += 1
-        time.sleep(15)
-
-
-if __name__ == '__main__':
-    # Call getSpice
-    getSpice()
+            self._unpack()
+
+        except KeyError:
+            print('SpiceyPy currently does not support your system.')
+
+        except RuntimeError as error:
+            print("CSPICE download failed with Error: {}!".format(error))
+
+    def _distribution_info(self):
+        """Creates the distribution name and the expected extension for the
+        CSPICE package and returns it.
+
+        Returns
+        -------
+        tuple
+           (distribution, extension) where distribution is the best guess from
+           the strings available within the platform_urls list of strings, and
+           extension is either "zip" or "tar.Z" depending on whether we are
+           dealing with a Windows platform or else.
+
+        Raises
+        ------
+        KeyError if the (system, machine) tuple does not correspond to any of
+                 the supported SpiceyPy environments.
+        """
+
+        print('Gathering information...')
+        system = platform.system()
+
+        # Cygwin system is CYGWIN-NT-xxx.
+        system = 'cygwin' if 'CYGWIN' in system else system
+
+        processor = platform.processor()
+        machine = '64bit' if sys.maxsize > 2 ** 32 else '32bit'
+
+        print('SYSTEM:   ', system)
+        print('PROCESSOR:', processor)
+        print('MACHINE:  ', machine)
+
+        return self._dists[(system, machine)]
+
+    def _download(self):
+        """Support function that encapsulates the OpenSSL transfer of the CSPICE
+        package to the self._local io.ByteIO stream.
+        """
+        self._local = _download_ssl(self._rcspice)
+
+    def _unpack(self):
+        """Unpacks the CSPICE package on the given root directory. Note that
+        Package could either be the zipfile.ZipFile class for Windows platforms
+        or tarfile.TarFile for other platforms.
+
+        Note
```

Comment:
not much of reason to keep the notes in the file, as long as the licenses and git history  are kept intact.